net/libreswan/Makefile

```diff
@@ -103,20 +122,82 @@ endef
 
 define Package/libreswan/install
 	$(INSTALL_DIR) \
-	 $(1)/etc/init.d \
 	 $(1)/etc/ipsec.d/policies \
 	 $(1)/usr/libexec/ipsec \
-	 $(1)/usr/sbin
+	 $(1)/usr/sbin \
+	 $(1)/etc/config \
+	 $(1)/etc/init.d \
+	 $(1)/etc/hotplug.d/libreswan \
+	 $(1)/etc/hotplug.d/iface \
+	 $(1)/usr/libexec/rpcd \
 
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/ipsec  \
 	    $(1)/usr/sbin/ipsec
-	$(INSTALL_BIN) ./files/ipsec.init $(1)/etc/init.d/ipsec
-	$(INSTALL_DATA) ./files/ipsec.conf $(1)/etc/ipsec.conf
-	$(INSTALL_DATA) ./files/ipsec.secrets $(1)/etc/ipsec.secrets
 	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/etc/ipsec.d/policies/* \
 	     $(1)/etc/ipsec.d/policies/
 	$(CP) $(PKG_INSTALL_DIR)/usr/libexec/ipsec/* \
 	    $(1)/usr/libexec/ipsec/
+
+	$(INSTALL_BIN) ./files/usr/libexec/ipsec/_updown.xfrm $(1)/usr/libexec/ipsec/_updown.xfrm
+	$(INSTALL_BIN) ./files/etc/init.d/ipsec $(1)/etc/init.d/ipsec
+	$(INSTALL_BIN) ./files/usr/libexec/rpcd/libreswan $(1)/usr/libexec/rpcd/libreswan
+	$(INSTALL_DATA) ./files/etc/ipsec.conf $(1)/etc/ipsec.conf
+	$(INSTALL_DATA) ./files/etc/ipsec.secrets $(1)/etc/ipsec.secrets
+	$(INSTALL_DATA) ./files/etc/config/libreswan $(1)/etc/config/libreswan
+	$(INSTALL_DATA) ./files/etc/hotplug.d/libreswan/01-user $(1)/etc/hotplug.d/libreswan/01-user
+	$(INSTALL_DATA) ./files/etc/hotplug.d/libreswan/02-vti $(1)/etc/hotplug.d/libreswan/02-vti
+	$(INSTALL_DATA) ./files/etc/hotplug.d/iface/89-libreswan $(1)/etc/hotplug.d/iface/89-libreswan
+endef
+
+define Package/libreswan-nftables
+	$(Package/libreswan/default)
+	TITLE+= nftables plugin)
+	DEPENDS+=firewall4 +libreswan +kmod-nft-xfrm +nftables \
+			 +kmod-nfnetlink-log
+endef
+
+define Package/libreswan-nftables/description
+	Provides Libreswan nftables plugin for adding firewall rules
+endef
+
+define Package/libreswan-nftables/install
+	$(INSTALL_DIR) $(1)/etc/hotplug.d/libreswan \
+		$(1)/usr/share/nftables.d/ruleset-post
+
+	$(CP) ./files/usr/share/nftables.d/* $(1)/usr/share/nftables.d
+	$(CP) ./files/etc/hotplug.d/libreswan/62-nftables $(1)/etc/hotplug.d/libreswan/62-nftables
+	$(LN) /tmp/libreswan/firewall.d/libreswan.rules $(1)/usr/share/nftables.d/ruleset-post/10_libreswan.nft
```

Comment:
I'm not sure about that now. But I would move the creation to `postinst` and add a `prerm` on package remove.  The `Package/libreswan-nftables/install` gets called on packaging on the buildsystem. I thing that using `$(1)` will not work there.

Reply:
Tested and verified
- bundled `libreswan-nftables` with firmware, it created a soft link in firmware
- removed package using `opkg remove libreswan-nftables`, It removed soft link
- installed package again using `opkg install libreswan-nftables` and it created software again.